### ISSUE TITLE
Can't auto import component as Pascal Case when use script lang="tsx" in SFC

### ISSUE THREAD

**Issue by chenweiyi** [nuxt/nuxt#26920]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.15.0`
- Nuxt Version:     `3.11.1`
- CLI Version:      `3.11.0`
- Nitro Version:    `2.9.4`
- Package Manager:  `pnpm@8.15.3`
- Builder:          `-`
- User Config:      `devtools`, `ssr`, `modules`, `unpluginHttpsReverseProxy`, `router`, `devServer`, `vite`
- Runtime Modules:  `@element-plus/nuxt@1.0.7`, `@unocss/nuxt@0.58.6`, `unplugin-https-reverse-proxy/nuxt`, `unplugin-icons/nuxt`, `@pinia/nuxt@0.5.1`, `dayjs-nuxt@2.1.9`
- Build Modules:    `-`

### Reproduction

![image](https://github.com/nuxt/nuxt/assets/15872986/0826170e-b47d-47b4-b174-c75d82468ec1)
![image](https://github.com/nuxt/nuxt/assets/15872986/a24db792-b80c-427f-8d51-18db31b26b37)

### Describe the bug

I use some element-plus component when I use tsx in SFC, but vscode ts-plugin error: ‘Can't find name "ElInput"'

when I manual import component like this `import { ElButton, ElInput, ElOption, ElSelect } from '#build/components';` , then It is work.

**Comment by chenweiyi** [comment]
reproduction url : https://github.com/chenweiyi/mini-nuxt-repo

Additional explanation:
1. When I reference the components of element-plus in the lang=tsx file using PascalCase, vscode ts prompts that the component does not exist, but it does not affect normal operation.
2. When I change the component from PascalCase to kebab-case in the lang=tsx file, there are no ts type errors, and the page works normally.

By the way, I would like to ask, in react tsx, component names need to be capitalized, but is there no such restriction in vue tsx?

![image](https://github.com/nuxt/nuxt/assets/15872986/6bc51a4b-46d7-4e3b-bd6e-6b59806b34f3)
![image](https://github.com/nuxt/nuxt/assets/15872986/67f7b0a5-3971-4422-8aa6-78130a3cede0)

**Comment by manniL** [comment]
@chenweiyi is this reproducible without any nuxt module too?

**Comment by chenweiyi** [comment]
Sorry, Do you mean this?

![image](https://github.com/nuxt/nuxt/assets/15872986/ed7cfb15-3e50-442d-8588-0f0c31a704c0)

**Comment by manniL** [comment]
@chenweiyi yes, please remove any modules not needed to reproduce the problem if possible ☺️

**Comment by chenweiyi** [comment]
Ok, I have now removed all the modules and unnecessary dependencies, but the issue still persists.

![image](https://github.com/nuxt/nuxt/assets/15872986/be65c134-b408-4fc3-89f9-1a6d0067038c)

**Comment by danielroe** [comment]
This seems to be a limitation of TSX - though I wonder if there is something Volar could do, or maybe within Vue's types?

- kebab-cased elements are handled as IntrinsicElements - we _could_ augment this to provide type safe access.

- pascal-cased elements are not resolved from the vue `GlobalComponents` interface. Even Vue global components are not resolved:
   ![CleanShot 2024-06-15 at 20 33 24@2x](https://github.com/nuxt/nuxt/assets/28706372/80fbca90-cbf3-465f-af44-840dee369cbc)

I think for now importing from `#components` is likely to be the safest approach. Alternatives could be both augmenting IntrinsicElements _and_ registering components as globals but I'm not sure that is a good idea.

cc: @antfu in case you have any other ideas.

**Comment by ElYaiko** [comment]
Any update?
Having to import the components manually is work 😕

**Comment by appli456** [comment]
I can only manually modify the tsconfig.json file to the form as shown in the picture so that the IDE can recognize the .tsx files. I'm not sure if this is a good solution. Perhaps the official side can deal with this issue.

![Image](https://github.com/user-attachments/assets/ba7e8d16-fc86-4a58-ac4e-cd90f122bb25)
